src/PlotlyPanel.tsx

```diff
@@ -129,15 +134,27 @@ export const PlotlyPanel: React.FC<Props> = (props) => {
   };
 
   const handlePlotRelayout = (event: Readonly<Plotly.PlotRelayoutEvent>) => {
-    const { "xaxis.range[0]": xAxisMin, "xaxis.range[1]": xAxisMax} = event;
+    const { "xaxis.range[0]": xAxisMin, "xaxis.range[1]": xAxisMax, "xaxis.autorange": autoRange } = event;
+
+    if (autoRange) {
+      props.onOptionsChange({...options, xAxis: { ...options.xAxis, min: undefined, max: undefined }});
+      return;
+    }
+
+    if (!xAxisMin || !xAxisMax) {
+      return;
+    }
 
     if (typeof xAxisMin === 'string' && typeof xAxisMax === 'string') {
       const from = dateTimeParse(xAxisMin);
       const to = dateTimeParse(xAxisMax);
 
       if (from.isValid() && to.isValid()) {
         props.onChangeTimeRange({ from: from.valueOf(), to: to.valueOf() });
+        props.onOptionsChange({...options, xAxis: { ...options.xAxis, min: from.valueOf(), max: to.valueOf() } });
       }
+    } else {
+      props.onOptionsChange({...options, xAxis: { ...options.xAxis, min: xAxisMin, max: xAxisMax } });
     }
   };
```

Comment:
This function handles updating the global time range when a user zooms in on a graph.
Zooming had some issues because it was a transient operation. As soon as the component re-renders, the axes would reset because the 'autorange' setting defaults to true. With this change, we persist the new X min and max to the panel's options to fix that problem.